### PR TITLE
Add declaration before code to guidelines

### PR DESCRIPTION
**Pull request by mkindahl**
Declarations before code has been informally supported and we have
several examples in the code. This commit adds that guideline to the
coding guidelines.

### CODE REVIEW

**Review by mfundul — APPROVED** [review]

**Comment by pmwkaa** [comment]
Is the idea to support it from now on?

**Comment by mkindahl** [comment]
> Is the idea to support it from now on?

We already have code in the repository with this, so nothing really changes. We are just documenting the guideline.

File: docs/StyleGuide.md [hunk 1]
```diff
@@ -37,6 +37,21 @@ elaborates on the situation. Instead, we've tried our best to develop
 a consistent style that roughly follows the style of the PostgreSQL
 source code.
 
+### Declarations before code
+
+C99 support having code before a declaration, similar to C++, and we
```
Comment (svenklemm) [inline docs/StyleGuide.md:42]: support => supports

**Review by svenklemm — APPROVED** [review]

File: docs/StyleGuide.md (hunk 1 above)
Comment (mkindahl) [reply docs/StyleGuide.md]: Fixed.

**Pull request merged**